.travis.yml

```diff
@@ -32,408 +41,156 @@ matrix:
-        - Rscript -e 'install.packages("covr"); Sys.setenv(NOT_CRAN = "true"); covr::codecov("r-pkg/")'
     - language: r
       warnings_are_errors: true
       cache: packages
       env:
         - ES_VERSION=5.0.2
         - TASK=rpkg
-      install:
-        - Rscript -e "devtools::install('r-pkg')"
-      script:
-        - R CMD build r-pkg/
-        - R CMD check *.tar.gz
-      after_success:
-        - Rscript -e 'install.packages("covr"); Sys.setenv(NOT_CRAN = "true"); covr::codecov("r-pkg/")'
     - language: r
       warnings_are_errors: true
       cache: packages
       env:
         - ES_VERSION=5.3.3
         - TASK=rpkg
-      install:
-        - Rscript -e "devtools::install('r-pkg')"
-      script:
-        - R CMD build r-pkg/
-        - R CMD check *.tar.gz
-      after_success:
-        - Rscript -e 'install.packages("covr"); Sys.setenv(NOT_CRAN = "true"); covr::codecov("r-pkg/")'
     - language: r
       warnings_are_errors: true
       cache: packages
       env:
         - ES_VERSION=5.4.3
         - TASK=rpkg
-      install:
-        - Rscript -e "devtools::install('r-pkg')"
-      script:
-        - R CMD build r-pkg/
-        - R CMD check *.tar.gz
-      after_success:
-        - Rscript -e 'install.packages("covr"); Sys.setenv(NOT_CRAN = "true"); covr::codecov("r-pkg/")'
     - language: r
       warnings_are_errors: true
       cache: packages
       env:
         - ES_VERSION=5.6.9
         - TASK=rpkg
-      install:
-        - Rscript -e "devtools::install('r-pkg')"
-      script:
-        - R CMD build r-pkg/
-        - R CMD check *.tar.gz
-      after_success:
-        - Rscript -e 'install.packages("covr"); Sys.setenv(NOT_CRAN = "true"); covr::codecov("r-pkg/")'
     - language: r
       warnings_are_errors: true
       cache: packages
       env:
         - ES_VERSION=6.0.1
         - TASK=rpkg
-      install:
-        - Rscript -e "devtools::install('r-pkg')"
-      script:
-        - R CMD build r-pkg/
-        - R CMD check *.tar.gz
-      after_success:
-        - Rscript -e 'install.packages("covr"); Sys.setenv(NOT_CRAN = "true"); covr::codecov("r-pkg/")'
     - language: r
       warnings_are_errors: true
       cache: packages
       env:
         - ES_VERSION=6.1.4
         - TASK=rpkg
-      install:
-        - Rscript -e "devtools::install('r-pkg')"
-      script:
-        - R CMD build r-pkg/
-        - R CMD check *.tar.gz
-      after_success:
-        - Rscript -e 'install.packages("covr"); Sys.setenv(NOT_CRAN = "true"); covr::codecov("r-pkg/")'
     - language: r
       warnings_are_errors: true
       cache: packages
       env:
         - ES_VERSION=6.2.4
         - TASK=rpkg
-      install:
-        - Rscript -e "devtools::install('r-pkg')"
-      script:
-        - R CMD build r-pkg/
-        - R CMD check *.tar.gz
       after_success:
-        - Rscript -e 'install.packages("covr"); Sys.setenv(NOT_CRAN = "true"); covr::codecov("r-pkg/")'
+        - .ci/report_to_covr.sh
```

Comment:
Maybe I'm misunderstanding the diff, but is there a reason you're only running the covr report on 6.2.4? Is it because you only want to run it once?

Reply:
Yeah it only needs to run once. So right now we're using `testthat` to run the tests, and then just reporting to `codecov` (an external service) one time to power the CI checks around coverage and the badge in the README.

Eventually I'd like to just use `covr` to run tests (like you and I have done in other projects) but didn't want to pull that into this PR.